packages/apps/plugins/plugin-thread/src/hooks/analytics.ts

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { useCallback } from 'react';
+
+import { ObservabilityAction } from '@braneframe/plugin-observability/meta';
+import { useIntentDispatcher } from '@dxos/app-framework';
+import { type ReactiveObject } from '@dxos/echo-schema';
+import { getSpace } from '@dxos/react-client/echo';
+import { useOnTransition } from '@dxos/react-ui';
+
+export const useAnalyticsCallback = (space: string | undefined, name: string, meta?: any) => {
+  const dispatch = useIntentDispatcher();
+
+  return useCallback(
+    (dynamicMeta?: any) => {
+      void dispatch({
+        action: ObservabilityAction.SEND_EVENT,
+        data: { name, ...meta, ...dynamicMeta, space },
```

Comment:
also all of the non-event name properties need to be inside `properties` field

Reply:
Ahh. Sweet.